sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/DecimalArithmeticOverrides.scala

```diff
@@ -99,21 +99,31 @@ object DecimalArithmeticOverrides {
             }
         }),
 
-      /**
-       * Because of https://github.com/NVIDIA/spark-rapids/issues/7595 we are not supporting
-       * Decimals for spark 3.4 and db 11.3. Once we do we should revert the changes made to the
-       * following tests test_mod, test_mod_mixed and test_mod_pmod_by_zero_not_ansi or we should
-       * just revert this commit
-       */
       expr[Remainder](
         "Remainder or modulo",
         ExprChecks.binaryProject(
           TypeSig.gpuNumeric, TypeSig.cpuNumeric,
-          ("lhs", TypeSig.integral + TypeSig.fp, TypeSig.cpuNumeric),
-          ("rhs", TypeSig.integral + TypeSig.fp, TypeSig.cpuNumeric)),
+          ("lhs", TypeSig.gpuNumeric, TypeSig.cpuNumeric),
+          ("rhs", TypeSig.gpuNumeric, TypeSig.cpuNumeric)),
```

Comment:
nit: it would be nice to add in a psNote for DECIMAL128 here to explain what we don't fully support.  This is really minor if we think we can get full remainder support in before we ship 23.06. (especially minor because we don't generate the support docs for anything but the oldest version of Spark that we support) Oh well....